app/src/main/java/com/markkolenbrander/capstonenewsapp/ArticleDataManager.kt

```diff
@@ -0,0 +1,32 @@
+package com.markkolenbrander.capstonenewsapp
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.preference.PreferenceManager
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import com.markkolenbrander.capstonenewsapp.models.Article
+
+class ArticleDataManager(application: Application) : AndroidViewModel(application){
+
+    private val context = application.applicationContext
+
+    private val sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
+    private val gson = Gson()
+
+    fun saveArticles(articles: ArrayList<Article?>){
+        val articlesJson = gson.toJson(articles)
+        sharedPrefs.edit().putString(PREFS_KEY_ARTICLES, articlesJson).apply()
+    }
+
+    fun fetchArticles(): ArrayList<Article?>{
+        val articleJson = sharedPrefs.getString(PREFS_KEY_ARTICLES, "")
+        val articleListTypeToken = object : TypeToken<ArrayList<Article?>>(){ }
+        return gson.fromJson(articleJson, articleListTypeToken.type)
+    }
```

Comment:
Looks like this function is the culprit of the crash for me.  You'll need to add a null check on the articlesJson first and return an empty ArrayList.

```suggestion
    fun fetchArticles(): ArrayList<Article?>{
        val articleJson = sharedPrefs.getString(PREFS_KEY_ARTICLES, "")
        
        if (articleJson.isNullOrBlank()) return ArrayList()
        
        val articleListTypeToken = object : TypeToken<ArrayList<Article?>>(){ }
        return gson.fromJson(articleJson, articleListTypeToken.type)
    }
```